modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionHash.scala

```diff
@@ -0,0 +1,21 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.api.TransactionIdentifierResponse
+import org.tessellation.rosetta.domain.metadata.GenericMetadata
+import org.tessellation.rosetta.domain.{NetworkIdentifier, TransactionIdentifier}
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.hex.Hex
+
+import derevo.circe.magnolia.customizableDecoder
+import derevo.derive
+
+case object ConstructionHash {
+  @derive(customizableDecoder)
+  case class Request(
+    networkIdentifier: NetworkIdentifier,
+    signedTransaction: Hex
+  )
+
+  def Response(hash: Hash, metadata: Option[GenericMetadata]) = TransactionIdentifierResponse(TransactionIdentifier(hash), metadata)
```

Comment:
Why uppercase def name?

Reply:
@TheMMaciek please see previous pull request, it matches the convention:
```
ConstructionDerive.Request
ConstructionDerive.Response

ConstructionHash.Request
ConstructionHash.Response
```

the only difference is that this response is an alias to another case class, so IMO it is correct

Reply:
All except `ContructionHash.Response` are actually case classes so the upper case is completely correct. Not sure about the def. Defs are normally lower case. Up to you guys.

Reply:
Yeah I get your point. To make it consistent we would need to:

```scala
case class Request(
    networkIdentifier: NetworkIdentifier,
    signedTransaction: Hex
)

def request = Request.apply _

def response(hash: Hash, metadata: Option[GenericMetadata]) = TransactionIdentifierResponse(TransactionIdentifier(hash), metadata)
```

Reply:
oooor in other way

```scala
object TransactionIdentifierResponse {
  def apply(hash: Hash, metadata: Option[GenericMetadata]) = TransactionIdentifierResponse(TransactionIdentifier(hash), metadata)
}
```

and then:

```scala
case object ConstructionHash {
   case class Request(
     networkIdentifier: NetworkIdentifier,
     signedTransaction: Hex
  )

   type Response = TransactionIdentifierResponse
   val Response = TransactionIdentifierResponse
}
```

Reply:
For me the second way is much nice because we won't have artificial method which is just `apply` and the API is still consistent.

Reply:
@TheMMaciek @kpudlik Sorry, but actually we are all wrong and the code itself as well @cngo-github.

According to the specification:
![image](https://user-images.githubusercontent.com/4959931/218118872-c50aa75f-040d-4e7c-b972-ad0f0ed0b936.png)

It should be:
```
@derive(customizableDecoder)
case class Request(
  networkIdentifier: NetworkIdentifier,
  signedTransaction: Hex
)

@derive(customizableEncoder)
case class Response(
  transactionIdentifier: TransactionIdentifier
)
```

The way how you did it @cngo-github  does not match the response from the rosetta specification.

Reply:
I like it. Looks really clean. And it's nice that you can have multiple apply methods and it works.

Reply:
I guess only `val Response` is needed in your example (we don't need `type Response` right?)

Reply:
Responses above referenced @kpudlik idea.

Reply:
Changed.